meta: discard const qualifier

Have you deactivated warnings on your setup?

```
/home/markus/Projekte/Elektra/current/src/libs/meta/meta.c: In function ‘elektraMetaArrayToKS’:
/home/markus/Projekte/Elektra/current/src/libs/meta/meta.c:966:23: warning: passing argument 2 of ‘ksAppendKey’ discards ‘const’ qualifier from pointer target type
  ksAppendKey (result, meta);
                       ^
In file included from /home/markus/Projekte/Elektra/current/src/libs/meta/meta.c:9:0:
/home/markus/Projekte/Elektra/build/src/include/kdb.h:225:9: note: expected ‘struct Key *’ but argument is of type ‘const struct Key *’
 ssize_t ksAppendKey(KeySet *ks, Key *toAppend);
```

--- a/src/libs/meta/meta.c
+++ b/src/libs/meta/meta.c
@@ -962,14 +962,14 @@ KeySet * elektraMetaArrayToKS (Key * key, const char * metaName)
 		ksRewind (result);
 		return result;
 	}
-	ksAppendKey (result, (Key *)meta);
+	ksAppendKey (result, keyDup (meta));
 	Key * currentKey = keyDup (meta);
 	keyAddName (currentKey, "#");
 	elektraArrayIncName (currentKey);
 	Key * curMeta = NULL;
 	while ((curMeta = (Key *)keyGetMeta (key, keyName (currentKey))) != NULL)
 	{
-		ksAppendKey (result, curMeta);
+		ksAppendKey (result, keyDup (curMeta));
 		elektraArrayIncName (currentKey);
 	}
 	keyDel (currentKey);

--- a/src/libs/meta/meta.c
+++ b/src/libs/meta/meta.c
@@ -962,7 +962,7 @@ KeySet * elektraMetaArrayToKS (Key * key, const char * metaName)
 		ksRewind (result);
 		return result;
 	}
-	ksAppendKey (result, meta);
+	ksAppendKey (result, (Key *)meta);
 	Key * currentKey = keyDup (meta);
 	keyAddName (currentKey, "#");
 	elektraArrayIncName (currentKey);

